src/core/Core.js

```diff
@@ -128,9 +128,9 @@ class Uppy {
    */
   setState (stateUpdate) {
     const newState = Object.assign({}, this.state, stateUpdate)
-    this.emit('core:state-update', this.state, newState, stateUpdate)
 
     this.state = newState
+    this.emit('core:state-update', this.state, newState, stateUpdate)
```

Comment:
`this.state` is the same as `newState ` at this point, so if we go this route, we have to save previous state, something like:

```js
const prevState = Object.assign({}, this.state)
this.state = newState
...
```